app/workers/bgs/submit_form686c_job.rb

```diff
@@ -11,30 +11,35 @@ class Invalid686cClaim < StandardError; end
 
     sidekiq_options retry: false
 
-    def perform(user_uuid, saved_claim_id, vet_info)
-      user = User.find(user_uuid)
-      Rails.logger.info('BGS::SubmitForm686cJob running!', { user_uuid:, saved_claim_id:, icn: user&.icn })
+    # rubocop:disable Metrics/MethodLength
+    def perform(user_uuid, icn, saved_claim_id, vet_info)
+      Rails.logger.info('BGS::SubmitForm686cJob running!', { user_uuid:, saved_claim_id:, icn: })
       in_progress_form = InProgressForm.find_by(form_id: FORM_ID, user_uuid:)
       in_progress_copy = in_progress_form_copy(in_progress_form)
       claim_data = valid_claim_data(saved_claim_id, vet_info)
       normalize_names_and_addresses!(claim_data)
-
-      BGS::Form686c.new(user).submit(claim_data)
+      user_struct = generate_user_struct(vet_info['veteran_information'])
```

Comment:
I think you could refactor the conditionals below by defining `user` as `feature_toggle ? user_struct : User.find` (pseudocode) and then leaving the rest of the code below intact. No need for that to block merge/deploy though! Up to you.